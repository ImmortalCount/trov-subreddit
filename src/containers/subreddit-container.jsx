import React, { Component } from 'react'
import { shape, func, instanceOf } from 'prop-types'
import { connect } from 'react-redux'

import { setArticles } from '../actions/reddit-data'
import { getSubreddits } from '../services/get-subreddits'
import UnformattedList from '../components/elements/unformatted-list'
import ArticleItem from '../components/article-item'

class SubredditContainer extends Component {
    static propTypes ={
        articles: instanceOf(Array).isRequired,
        match: shape({}).isRequired,
        dispatch: func.isRequired
    }

    componentDidMount() {
        getSubreddits(this.props.match.params.sub).then((res) => {
            this.props.dispatch(
                setArticles(
                    res.data.data.children,
                    this.props.match.params.sub
                )
            )
        })
    }

    render() {
        return (
            <UnformattedList>
                {this.props.articles.map(
                    item => (
                        (
                            <ArticleItem
                                data={item.data}
                                key={item.data.name}
                            />
                        )
                    )
                )}
            </UnformattedList>
        )
    }
}

const mapDispatchToProps = state => ({
    articles: state.redditData.articles
})

export default connect(mapDispatchToProps)(SubredditContainer)
