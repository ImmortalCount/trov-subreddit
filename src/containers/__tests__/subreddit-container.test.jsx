import React from 'react'
import { shallow } from 'enzyme'

import { testData } from '../../utils/test-data'
import { SubredditContainer } from '../subreddit-container'

describe('SubredditContainer Component', () => {
    const defaultProps = {
        articles: [testData],
        match: {
            params: {
                sub: 'guitarpedals'
            }
        },
        dispatch: jest.fn()
    }

    const wrapper = shallow(
        <SubredditContainer {...defaultProps} />
    )

    it('should render SubredditContainer component', () => {
        expect(wrapper.find('UnformattedList').length).toBe(1)
        expect(wrapper.find('Button').length).toBe(1)
    })

    it('should test intance methods', () => {
        const instance = shallow(
            <SubredditContainer {...defaultProps} />
        ).instance()
        instance.componentDidUpdate()
        instance.componentWillUnmount()
        instance.getNewArticles()
        instance.fetchAndStoreSubreddits()
        instance.props.dispatch()
        expect(instance.props.dispatch).toBeCalled()
    })
})
