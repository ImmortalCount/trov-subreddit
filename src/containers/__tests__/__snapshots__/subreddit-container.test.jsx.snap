// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubredditContainer Component SubredditContainer matches snapshot 1`] = `
Array [
  <ul
    className="sc-bdVaJa EFgaU"
  >
    <li
      className="sc-htpNat IGjmT"
    >
      <span
        className="sc-bwzfXH benIyF"
      >
        no image
      </span>
      <div
        className="sc-ifAKCX cFlEyZ"
      >
        <h2>
          <a
            href="https://www.reddit.com//r/guitarpedals/comments/aibcl4/namm_2019_hype_thread/"
            rel="noopener noreferrer"
            target="_blank"
          >
            NAMM 2019 Hype Thread
          </a>
        </h2>
        <p>
          Posted by 
           
          <a
            href="https://www.reddit.com/user/slap_me_thrice"
            rel="noopener noreferrer"
            target="_blank"
          >
            slap_me_thrice
          </a>
           
          8 days
           
           ago
        </p>
        <p>
          <a
            href="https://www.reddit.com//r/guitarpedals/comments/aibcl4/namm_2019_hype_thread/#siteTable"
            rel="noopener noreferrer"
            target="_blank"
          >
            comments
          </a>
        </p>
      </div>
    </li>
  </ul>,
  <button
    className="sc-EHOje ixXjDj"
    color="green"
    onClick={[Function]}
    size="large"
  >
    Get new articles
  </button>,
]
`;
